### DISCUSSION TITLE
To pass view name as paramater

### DISCUSSION THREAD

**Discussion by sreejithkalarikkal** [sqlpage/SQLPage#567] [Q&A]
Is it possible to pass a view name as paramater?The sample code is written below.

```ts
@ipNav({
    caption: "Evidence Inner List",
    description: ``,
    siblingOrder: 3,
  })
  "infra/policy/evidence_details.sql"() {
    return this.SQL`
      ${this.activePageTitle()}
      select 'table'as component;
      **select * from $viewname::TEXT**

      `;
  }
 ```
 
 I am passing **viewname**  as link from the paranet page and i need to display the content of view.I tried the above way but it is not working.Anyone please help on this.

**Comment by lovasoa** [comment]
Hello and welcome to SQLPage ! I am not familiar with the syntax you are using. This seems to be typescript with some decorator library ? Can you tell us a bit more about the context ? 

About using variables for table names in sqlpage: this is not supported, because it should be avoided.
See this other question about the same topic: https://github.com/lovasoa/SQLpage/discussions/518

So, in short: if you really insist, you can make this work, but I would strongly recommend against it. If you have a set of views that you want to expose, I would recommend hardcoding their names in your application.

**Reply by lovasoa** [reply]
If you want to ignore my advice and still allow random access to tables to users, here is a postgres function you can use:

```sql
CREATE OR REPLACE FUNCTION get_table_as_json_set(table_name text) 
RETURNS SETOF json AS $$
DECLARE
    query text;
BEGIN
    query := format('SELECT row_to_json(t) FROM %I t', table_name);
    RETURN QUERY EXECUTE query;
END;
$$ LANGUAGE plpgsql;
```

and in your page

```sql
select 'table' as component;
select 'dynamic' as component, properties from get_table_as_json_set($viewname) as t(properties);
```

**Comment by sreejithkalarikkal** [comment]
Please find the attached screen shot(1.png
![1](https://github.com/user-attachments/assets/e6235b01-15de-470f-a532-7182a23d4c65)
![2](https://github.com/user-attachments/assets/37f2e4fe-a6c2-41f6-b255-a88129872c32)
) here I am passing a view  **viup_time** (see the bottom left).When I click on the card Up time(See top left) the this view need to be passed to the child component .Here I  cant hard code all the view since the data in the cards are redirecting to different view.I need to write it as a common way.I am also attaching the error(2.png) when I click on the card data.

**Reply by lovasoa** [reply]
See my response above. I would recommend creating multiple pages instead of a single evidence_details.sql. You could have a `viup_time.sql` file with the plain table name in it.

**Comment by sreejithkalarikkal** [comment]
So I have 100 cards with different views you mean I need to make 100 .sql file ?Is it a pretty good idea?
If any possibility of making it a dynamic method we can reduce the code .

**Reply by lovasoa** [reply]
Yes, that's what I am saying. But have you really written 100 unique views ?

**Comment by sreejithkalarikkal** [comment]
Yes it may be 100 or more, for testing purpose I have only displayed some of it. This data are for audit purpose ,so that the auditor can easily check each of the data.

**Reply by lovasoa** [reply]
If you really want to allow dynamic access to any table in your database; see my first answer above. But that is not something I would recommend.
